### PR TITLE
Add cross-env package

### PR DESCRIPTION
**Pull request by NavizDev**
Most Windows command prompts will choke when you set environment variables with `NODE_ENV=development`
image

So, I have added the cross-env package for a multiplatform solution

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| buildDuration | 12.7s | 12.7s | -26ms |
| nodeModulesSize | 58.8 MB | 58.8 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.24 | 2.189 | -0.05 |
| / avg req/sec | 1115.99 | 1142.27 | ⚠️ +26.28 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.451 | 1.394 | -0.06 |
| /error-in-render avg req/sec | 1723.04 | 1793.22 | ⚠️ +70.18 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.23 kB | 6.23 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5232.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.4 kB | 56.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.83 kB | 4.83 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.88 kB | 6.88 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.6 kB | 51.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error.js gzip | 3.13 kB | 3.13 kB | ✓ |
| hooks.js gzip | 663 B | 663 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.module.js gzip | 598 B | 598 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| index.html gzip | 926 B | 926 B | ✓ |
| link.html gzip | 936 B | 936 B | ✓ |
| withRouter.html gzip | 925 B | 925 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| buildDuration | 13.6s | 13.9s | ⚠️ +264ms |
| nodeModulesSize | 58.8 MB | 58.8 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.23 kB | 6.23 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5232.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.4 kB | 56.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.83 kB | 4.83 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.88 kB | 6.88 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.6 kB | 51.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error.js gzip | 3.13 kB | 3.13 kB | ✓ |
| hooks.js gzip | 663 B | 663 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.module.js gzip | 598 B | 598 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _error.js | 853 kB | 853 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 853 kB | 853 kB | ✓ |
| link.js | 891 kB | 891 kB | ✓ |
| routerDirect.js | 883 kB | 883 kB | ✓ |
| withRouter.js | 883 kB | 883 kB | ✓ |
| Overall change | 4.37 MB | 4.37 MB | ✓ |

</details>

</details>

File: examples/with-firebase-authentication-serverless/README.md [hunk 1]
```diff
@@ -45,3 +45,7 @@ After deploying, copy the deployment URL and navigate to your Firebase project's
 ## The idea behind the example
 
 This example includes Firebase authentication and serverless [API routes](https://nextjs.org/docs/api-routes/introduction). On login, the app calls `/api/login`, which stores the user's info (their decoded Firebase token) in a cookie so that it's available server-side in `getInitialProps`. On logout, the app calls `/api/logout` to destroy the cookie.
+
+## Notes
+
+- **(Optional)** if you use windows os, install `npm install -g win-node-env` to avoid having future problems with `NODE_ENV = development` in `package.json`.
```
Comment (lfades) [inline examples/with-firebase-authentication-serverless/README.md:51]:
```suggestion
```

File: examples/with-firebase-authentication-serverless/package.json [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
-    "dev": "NODE_ENV=development next dev",
-    "build": "NODE_ENV=production next build",
-    "start": "NODE_ENV=production next start"
+    "dev": "cross-env NODE_ENV=development next dev",
+    "build": "cross-env NODE_ENV=production next build",
+    "start": "cross-env NODE_ENV=production next start"
   },
   "dependencies": {
     "cookie-session": "1.4.0",
+    "cross-env": "^7.0.2",
```
Comment (lfades) [inline examples/with-firebase-authentication-serverless/package.json:11]: Can you move it to `devDependencies`

**Comment by lfades** [comment]
Wait, why is setting `NODE_ENV` required in this example at all? Next.js should take care of it :thinking:

File: examples/with-firebase-authentication-serverless/README.md (hunk 1 above)
Comment (NavizDev) [reply examples/with-firebase-authentication-serverless/README.md:51]: wow! 😥  I was excited and sent those lines by mistake-

**Review by NavizDev — COMMENTED** [review]
> Wait, why is setting `NODE_ENV` required in this example at all? Next.js should take care of it 🤔
> 
> It should work if you run `next dev` - `next build` and `next start` without it

because, Windows users have the error with NODE_ENV

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| buildDuration | 11.5s | 11.5s | -22ms |
| nodeModulesSize | 58.8 MB | 58.8 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.09 | 1.997 | -0.09 |
| / avg req/sec | 1196.14 | 1252.14 | ⚠️ +56 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.354 | 1.267 | -0.09 |
| /error-in-render avg req/sec | 1846.51 | 1973.3 | ⚠️ +126.79 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.23 kB | 6.23 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5232.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.4 kB | 56.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.83 kB | 4.83 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.88 kB | 6.88 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.6 kB | 51.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error.js gzip | 3.13 kB | 3.13 kB | ✓ |
| hooks.js gzip | 663 B | 663 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.module.js gzip | 598 B | 598 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| index.html gzip | 926 B | 926 B | ✓ |
| link.html gzip | 936 B | 936 B | ✓ |
| withRouter.html gzip | 925 B | 925 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| buildDuration | 12.6s | 12.3s | -218ms |
| nodeModulesSize | 58.8 MB | 58.8 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.23 kB | 6.23 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..5232.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.4 kB | 56.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.83 kB | 4.83 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.88 kB | 6.88 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 51.6 kB | 51.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error.js gzip | 3.13 kB | 3.13 kB | ✓ |
| hooks.js gzip | 663 B | 663 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.86 kB | 7.86 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _app.module.js gzip | 598 B | 598 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | NavizDev/next.js add-cross-env | Change |
| - | - | - | - |
| _error.js | 853 kB | 853 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 853 kB | 853 kB | ✓ |
| link.js | 891 kB | 891 kB | ✓ |
| routerDirect.js | 883 kB | 883 kB | ✓ |
| withRouter.js | 883 kB | 883 kB | ✓ |
| Overall change | 4.37 MB | 4.37 MB | ✓ |

</details>

</details>

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Thanks!

**Comment by timneutkens** [comment]
I'd like to get rid of NODE_ENV in this example like @lfades said though, it's not needed indeed.

**Pull request merged**